src/familiar.ts

```diff
@@ -55,177 +53,237 @@ export function meatFamiliar(): Familiar {
   return _meatFamiliar;
 }
 
-function myFamiliarWeight(familiar: Familiar | null = null) {
-  if (familiar === null) familiar = myFamiliar();
-  return familiarWeight(familiar) + weightAdjustment();
+type GeneralFamiliar = {
+  familiar: Familiar;
+  expectedValue: number;
+};
+
+type StandardDropFamiliar = {
+  familiar: Familiar;
+  expected: number[];
+  drop: Item;
+  pref: propertyTypes.NumericProperty;
+  additionalValue?: number;
+};
+
+function valueStandardDropFamiliar({
+  familiar,
+  expected,
+  drop,
+  pref,
+  additionalValue,
+}: StandardDropFamiliar): GeneralFamiliar | null {
+  const expectedTurns = expected[get(pref)];
```

Comment:
If we get more than `expected.length` drops this would overflow the memory
```suggestion
  const expectedTurns = expected[Math.min(expected.length - 1, get(pref))];
```

Reply:
This is not a necessary change; line 77 checks to see if expectedTurns is defined. Overflowing an array returns undefined.